armi/reactor/blocks.py

```diff
@@ -1929,6 +1938,48 @@ def _getPinCoordinatesHex(self):
                 coordinates.append(xyz)
         return coordinates
 
+    def autoCreateSpatialGrids(self):
+        """
+        Given a block without a spatialGrid, create a spatialGrid and give its children
+        the corresponding spatialLocators (if it is a simple block). In this case, a simple block would
+        be one that has either multiplicity of components equal to 1 or n but no other multiplicities.
```

Comment:
...where N fits exactly into a given number of hex rings. I think this is pretty critical. We should also be verifying this in the implementation and throw a ValueError if it's not satisfied.